File: solutions/hash_table/1-2.py
"""
Problem Number 1.2
Problem : Given two string, decide if one is a permutation of the other one.
"""

import array

class permutation(object):

	def __init__(self):
		self.hash_for_s1 = [0]*128
		self.hash_for_s2 = [0]*128

	def check_permutation(self):
		
		print ("Please enter the first string : ")
		s1=input()

		print("Please enter the second string : ")
		s2=input()

		if len(s1) != len(s2) :
			print ("Not permutation to each other")
			return

		for i in s1:
			self.hash_for_s1[ord(i)] =  self.hash_for_s1[ord(i)] + 1

		for i in s2:
			self.hash_for_s2[ord(i)] = self.hash_for_s2[ord(i)] + 1 

		for i in range (0,127):
			if self.hash_for_s1[i] != self.hash_for_s2[i] :
				print ("Not permutation to each other")
				return 
		print ("Permutation to each other")


p = permutation()
p.check_permutation()

File: solutions/hash_table/1-2.cpp
/*
Problem Number 1.2
Problem : Givem two string, decide if one is a permutation of the other one.
*/

#include <iostream>
#include <string>

using namespace std;

class permutation
{
	string s1, s2;
	int HashForS1[128] = { 0 };
	int HashForS2[128] = { 0 };

	public: 
	void checkPermutation()
	{
		cout << "\nPlease enter the first string : ";
		getline(std::cin, s1);

		cout << "\nPlease enter the second string : ";
		getline(std::cin, s2);

		if (s1.size() != s2.size())
		{
			cout << "\nNot permutation to each other";
			return;
		}

		for (int i = 0; i <s1.size(); i++)
		{
			HashForS1[(int(s1[i]))] = HashForS1[(int(s1[i]))] + 1;
		}
		 
		for (int i = 0; i < s2.size(); i++)
		{
			HashForS2[(int(s2[i]))] = HashForS2[(int(s2[i]))] + 1;
		}
	 
		for (int i = 0; i < 128 ; i++)
		{
			if (HashForS1[i] != HashForS2[i])
			{
				cout << "\nNot permutation to each other";
				return;
			}
		}
		cout << "Permutation to each other";
	}
};

int  main()
{
	permutation permutation_obj;
	permutation_obj.checkPermutation();
	
}
